src/action.ts

```diff
@@ -2,55 +2,75 @@ import * as core from '@actions/core'
 import * as util from 'util'
 import * as child_process from 'child_process'
 
-const { GITHUB_ACTOR } = process.env
+class GitArgs {
+  readonly command: string
+
+  constructor(
+    readonly ref: string,
+    readonly directory: string,
+  ) {
+    this.command = `git -C ${this.directory}`
+  }
+}
 
 async function exec(command: string) {
   const { stdout, stderr } = await util.promisify(child_process.exec)(command)
   if (stderr) console.error(stderr)
   return stdout
 }
 
-function annotatedTag(message: string, ref: string) {
+function annotatedTag(message: string, git: GitArgs) {
   core.info('Creating annotated tag...')
-  return exec(`git tag -a -f -m "${message}" ${ref}`)
+  return exec(`${git.command} tag -a -f -m "${message}" ${git.ref}`)
 }
 
-function lightweightTag(ref: string) {
+function lightweightTag(git: GitArgs) {
   core.info('Creating lightweight tag...')
-  return exec(`git tag -f ${ref}`)
+  return exec(`${git.command} tag -f ${git.ref}`)
 }
 
-function forceBranch(ref: string) {
+function forceBranch(git: GitArgs) {
   core.info('Updating branch...')
-  return exec(`git branch -f ${ref}`)
+  return exec(`${git.command} branch -f ${git.ref}`)
 }
 
-async function run() {
-  try {
+function setupUser(git: GitArgs) {
     core.info('Setting up git user...')
-    await exec(`git config user.name "${GITHUB_ACTOR}"`)
+
+    const { GITHUB_ACTOR } = process.env
+
+    await exec(`${git.command} config user.name "${GITHUB_ACTOR}"`)
     await exec(
-      `git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"`
+      `${git.command} config user.email "${GITHUB_ACTOR}@users.noreply.github.com"`
     )
+}
 
-    const message = core.getInput('description')
-    const ref = core.getInput('ref') || core.getInput('tag-name') || 'latest'
-    core.info(`Using '${ref}' as tag name.`)
+async function run() {
+  try {
+    const git = new GitArgs(
+      core.getInput('git-directory'),
+      core.getInput('ref') || core.getInput('tag-name') || 'latest')
 
     const branch = core.getBooleanInput('force-branch', { required: true })
+    const message = core.getInput('description')
 
     if (branch && message)
       core.warning(
         "You can't set a message when updating a branch, the message will be ignored."
       )
 
-    if (branch) await forceBranch(ref)
-    else if (message) await annotatedTag(message, ref)
-    else await lightweightTag(ref)
+    core.info(`Running git commands within ${git.directory}`)
+    core.info(`Using '${git.ref}' as tag name.`)
+
+    setupUser(git)
+
+    if (branch) await forceBranch(git)
+    else if (message) await annotatedTag(message, git)
+    else await lightweightTag(git)
```

Comment:
Rearranged some of the messaging. Happy to revert if it will hold back the PR.

Reply:
Nice!